### ISSUE TITLE
Convert JS stories to MDX

### ISSUE THREAD

**Issue by dhruvkb** [WordPress/openverse#584]
## Problem

JS stories have many shortcomings compared to MDX. While MDX stories can be cumbersome, they really shine when it comes to being able to write long-form documentation, including helpful tips about the props, functionality and expected use-cases of components. In a sense, they're almost equivalent to having a `README.md` for every component.

This has been spun-off from [this discussion](https://github.com/WordPress/openverse-frontend/discussions/465#discussioncomment-1743808).

## Description

The goal is to convert the existing JS stories to their MDX counterparts. To keep PRs reviewable, please do only one or two components at a time.

## Alternatives

We can continue to use a mix of JS and MDX stories as currently done. This is bad because why would you mix two ways of doing one thing in one project.

The other alternative is to go the other route (aka backwards) and convert all MDX stories to the less flexible JS format. This would add consistency but at a huge price.

## Additional context

There are a lot of MDX stories in the repo to refer to. Just pick the longest, more comprehensive one as a reference.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

**Comment by sarayourfriend** [comment]
@sepehrrezaei Apologies, but I think this issue might need some additional ticket work to decide how the project best wants to solve it.

@dhruvkb and @obulat do y'all have any objection to the approach I shared above? I do think it's important that the stories are type checked.

**Comment by dhruvkb** [comment]
@sarayourfriend I still prefer MDX because of the following reasons.

- The Docs page can actually be long-form docs with some supporting stories mixed into it, which is the opposite of your suggestion where each story has some documentation mixed into it. This approach will leave the Docs page empty.
- MDX keeps the documentation outside the story, so the code for the story is concerned purely with the component while this approach will leak documentation into the individual stories tab.
- I find the built-in doc-page components such as `ArgsTable`, `Description` etc. seem very useful and worth keeping around. They cannot be used if the documentation is written inside `.vue` components.
- Typechecking of stories, while important, seems like something I can compromise on and my mind would be at ease because stories have (or should have) VR tests and the rest of the app has other tests (unit, e2e and VR).
- I actually consider MDX to be amazingly straightforward, except for some boilerplate in the start to set things up. The long-form docs aspect of MDX is purely Markdown, that anyone can read and write fairly easily.

To summarise, personally, I favour MDX and am willing to lose type safety in the stories to use it. _However, if you could make a PR, we can all see how the end result looks (from the PR preview) and make a final call on what works._

**Comment by sepehrrezaei** [comment]
@sarayourfriend & @dhruvkb So after your discussions, Can I work on it ?

**Comment by sarayourfriend** [comment]
@sepehrrezaei Yup! I've assigned the issue to you :+1: Thanks for hopping in.
